internal/component/postgresql/postgresdbaccess.go

```diff
@@ -329,7 +329,7 @@ func (p *PostgresDBAccess) doDelete(parentCtx context.Context, db dbquerier, req
 
 	var result pgconn.CommandTag
 	if req.ETag == nil || *req.ETag == "" {
-		result, err = db.Exec(parentCtx, "DELETE FROM state WHERE key = $1", req.Key)
+		result, err = db.Exec(parentCtx, "DELETE FROM "+p.metadata.TableName+" WHERE key = $1", req.Key)
```

Comment:
Per conversation here, the table name is sanitized so cannot cause a SQL injection. Looks good as a result.